src/test/packages/22-git-data/zarf.yaml

```diff
@@ -28,7 +28,7 @@ components:
       onDeploy:
         before:
           # Check to verify the package secret has been saved for the already deployed component
-          - cmd: test $(./zarf tools kubectl get secret -n zarf zarf-package-git-data-test -o jsonpath='{.data.*}' | base64 --decode | jq -r .deployedComponents | jq '. | length') -eq 1
+          - cmd: test $(./zarf tools kubectl get secret -n zarf zarf-package-git-data-test -o jsonpath='{.data.*}' | base64 --decode | jq -r .deployedComponents | jq '. | length') -eq 2
```

Comment:
This seems weird/wrong?  Maybe due to the weird installed chart logic?

Reply:
This is not due to the installedCharts logic.

This PR changes when and how frequently Zarf updates package secrets when deploying package components.

Zarf adds a component to `deployedComponents` and updates the package secret with this data prior to actually deploying the component. So in this case, Zarf has marked this component as `Deploying` in the package secret, resulting in the increased amount of `deployedComponents` in the package secret here.